packages/cli/src/plugin.js

```diff
@@ -0,0 +1,228 @@
+import fs from 'fs'
+import path from 'path'
+
+import chalk from 'chalk'
+
+import { getConfig, getPaths } from './lib'
+import { installModule } from './lib/packages'
+
+const PLUGIN_CACHE_FILENAME = 'command-cache.json'
+
+function validatePluginList(plugins) {
+  // Plugins should only occur once in the list
+  const pluginNames = plugins.map((p) => p.package)
+  if (plugins.length !== new Set(pluginNames).size) {
+    console.warn(
+      chalk.yellow(
+        '⚠️  Duplicate plugin names found in redwood.toml, duplicates will be ignored.'
+      )
+    )
+  }
+
+  // Plugins should be published to npm under a scope which is used as the namespace
+  const namespaces = plugins.map((p) => p.package.split('/')[0])
+  namespaces.forEach((ns) => {
+    if (!ns.includes('@')) {
+      console.warn(
+        chalk.yellow(
+          `⚠️  Plugin "${ns}" is missing a scope/namespace, it will not be loaded.`
+        )
+      )
+    }
+  })
+}
+
+export async function loadPlugins(yargs) {
+  const { plugins, autoInstall } = getConfig().cli
+
+  // TODO: Remove this when we make the full switch to a total plugin based CLI
+  // If we don't have any plugins in the toml file, add all the current @redwoodjs plugins
+  if (plugins.length === 0) {
+    // TODO: Start splitting up CLI into plugins
+    // Nothing to add here at the moment
+  }
+
+  // Validate plugin list from redwood.toml, will print warnings to the console
+  validatePluginList(plugins)
+
+  // Get a list of all unique namespaces, sorted alphabetically but with @redwoodjs first
+  // we ignore any invalid names i.e. names without an '@'
+  const namespaces = Array.from(
+    new Set([
+      ...plugins
+        .filter((p) => p.package.startsWith('@redwoodjs/'))
+        .map((p) => p.package.split('/')[0])
+        .sort((a, b) =>
+          a.package > b.package ? 1 : b.package > a.package ? -1 : 0
+        ),
+      ...plugins
+        .filter((p) => !p.package.startsWith('@redwoodjs/'))
+        .map((p) => p.package.split('/')[0])
+        .sort((a, b) =>
+          a.package > b.package ? 1 : b.package > a.package ? -1 : 0
+        ),
+    ])
+  ).filter((ns) => ns.includes('@'))
```

Comment:
Reason to use one or the other? The scope/namespace always comes first right?

```suggestion
  ).filter((ns) => ns.startsWith('@'))
```